sources/packages/backend/libs/integrations/src/institution-integration/ier12-integration/ier12-file-detail.ts

```diff
@@ -1,114 +1,578 @@
+    record.appendWithStartFiller(this.booksAndSuppliesCost, 10, NUMBER_FILLER);
+    record.appendWithStartFiller(this.mandatoryFees, 10, NUMBER_FILLER);
+    record.appendWithStartFiller(this.exceptionExpenses, 10, NUMBER_FILLER);
+    record.appendWithStartFiller(this.totalFundedWeeks, 2, NUMBER_FILLER);
+    record.append(this.courseLoad, 3);
+    record.append(this.offeringIntensityIndicator, 1);
+    record.appendDate(this.applicationSubmittedDate, DATE_FORMAT);
+    record.append(this.programYear, 8);
+    record.append(this.applicationStatusCode, 4);
+    record.appendDate(this.applicationStatusDate, DATE_FORMAT);
+    record.appendWithStartFiller(this.cslAmount, 10, NUMBER_FILLER);
+    record.appendWithStartFiller(this.bcslAmount, 10, NUMBER_FILLER);
+    record.appendWithStartFiller(this.epAmount, 10, NUMBER_FILLER);
+    record.append(this.provincialDefaultFlag, 1);
+    record.append(emptyStringFiller(this.federalDefaultFlag, SPACE_FILLER), 1);
+    record.append(this.provincialOverawardFlag, 1);
+    record.append(this.federalOverawardFlag, 1);
+    record.append(this.restrictionFlag, 1);
+    record.appendDate(this.scholasticStandingEffectiveDate, DATE_FORMAT, {
+      filler: SPACE_FILLER,
+    });
+    record.appendWithEndFiller(
+      emptyStringFiller(this.scholasticStandingCode),
+      4,
+      SPACE_FILLER,
+    );
+    record.appendDate(this.assessmentDate, DATE_FORMAT);
+    record.appendDate(this.withdrawalDate, DATE_FORMAT, {
+      filler: SPACE_FILLER,
+    });
+    record.appendWithStartFiller(
+      emptyNumberFiller(this.applicantAndPartnerExpectedContribution),
+      10,
+      NUMBER_FILLER,
+    );
+    record.appendWithStartFiller(
+      emptyNumberFiller(this.parentExpectedContribution),
+      10,
+      NUMBER_FILLER,
+    );
+    record.appendWithStartFiller(
+      emptyNumberFiller(this.totalExpectedContribution),
+      10,
+      NUMBER_FILLER,
+    );
+    record.appendWithStartFiller(
+      emptyNumberFiller(this.dependantChildQuantity),
+      3,
+      NUMBER_FILLER,
+    );
+    record.appendWithStartFiller(
+      emptyNumberFiller(this.dependantChildInDaycareQuantity),
+      3,
+      NUMBER_FILLER,
+    );
+    record.appendWithStartFiller(
+      emptyNumberFiller(this.dependentInfantQuantity),
+      3,
+      NUMBER_FILLER,
+    );
+    record.appendWithStartFiller(
+      emptyNumberFiller(this.dependantOtherQuantity),
+      3,
+      NUMBER_FILLER,
+    );
+    record.appendWithStartFiller(
+      emptyNumberFiller(this.dependantPostSecondaryQuantity),
+      3,
+      NUMBER_FILLER,
+    );
+    record.appendWithStartFiller(
+      emptyNumberFiller(this.totalDependantQuantity),
+      3,
+      NUMBER_FILLER,
+    );
+    record.appendWithStartFiller(
+      emptyNumberFiller(this.familyMembersQuantity),
+      3,
+      NUMBER_FILLER,
+    );
+    record.append(emptyStringFiller(this.parent1Flag, SPACE_FILLER), 1);
```

Comment:
There is no benefit other than Sonar complaining about using `??` condition 65 times in the method. 

This IER is a peculiar type of file which has a lot of non mandatory fields and on top of that a huge chunk of fields without analysis done so I had to create a method to handle emptyString and emptyNumber to get through sonar

Reply:
I made this suggestion about `emptyStringFiller`. It is practically the same amount of code and it would make the code even cleaner.
https://github.com/bcgov/SIMS/pull/2246#discussion_r1316395596